petrussola/spacex
# Display local launch pad and time stamp from User locale when hoverin…

## Overview 

Reference story #XYZ

- Changing display time and date in launch detail tile from User locale into local time and date at launch pad.
- Display time and date at User locale when hovering over the timestamp.

## Update Part 1 - display pad local time

- After extensive research JS doesn't seem to offer methods to handle time zones - date objects are created in UTC.
- A number of options via 3rd party libraries may be available but I settled on implementing this feature via native JS.
- Created 2 new helper methods in format-date.js file: 1) transformAmPm and 2) checkDayPadTime.

1) While it would have been easier to parse time as it comes back from API, it didn't look consistent with the date object in User locale, which is in 12h format (AM/PM). This method returns the hour in this format alongside AM or PM.
2) Given that time pads are located in different time zones and that User locale is unknown, days in pad time and user time may be different. Since I am building pad time via parsing API response, this method returns the correct day at pad time (e.g. /launches/107 launch time at pad time was at ~7pm and my own locale (GMT) is past midnight the day after. 

## Update Part 2 - display User Local time as a tooltip when hovering over the date

- Used TooTip component from Chakra - https://chakra-ui.com/docs/overlay/tooltip
- A number of options are available, I settled on displaying it in default gray color, arrow form and at the bottom right of the space as it didn't interfere or overlaid with any other element of the page. Happy to be implement better looking options if design team wants to do so!

## How to test

Head to /launches, click on any launch. Check date under "Launch date" section and hover over to see the time stamp in User locale (based on browser date/time settings).
